packages/core/src/routes/resource.ts

```diff
@@ -0,0 +1,106 @@
+import { Resources } from '@logto/schemas';
+import { object, string } from 'zod';
+
+import koaGuard from '@/middleware/koa-guard';
+import koaPagination from '@/middleware/koa-pagination';
+import {
+  findTotalNumberOfResources,
+  findAllResources,
+  findResourceById,
+  insertResource,
+  updateResourceById,
+  deleteResourceById,
+} from '@/queries/resource';
+import { findAllScopesWithResourceId } from '@/queries/scope';
+import { buildIdGenerator } from '@/utils/id';
+
+import { AuthedRouter } from './types';
+
+const resourceId = buildIdGenerator(21);
+
+export default function resourceRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/resources', koaPagination(), async (ctx, next) => {
+    const { limit, offset } = ctx.pagination;
+
+    const [{ count }, resources] = await Promise.all([
+      findTotalNumberOfResources(),
+      findAllResources(limit, offset),
+    ]);
+
+    // Return totalCount to pagination middleware
+    ctx.pagination.totalCount = count;
```

Comment:
Per koaPagination design, if `totalCount` is not provided our custom pagination headers won't be set in the response.  But it does not throw any exceptions. cc: @wangsijie

Reply:
Since `koaPagination` is a global middleware, `totalCount` existancy is used to detect whether pagination is needed in this endpoint.
We can do this after `koaPagination` becomes route-specific.

Reply:
1. let's add an issue for throw error after using `koaPagination` without setting `ctx.pagination.totalCount`.
2. if it's a global middleware, should we use it in the beginning of the middleware chain instead of per-route?

Reply:
LOG-1336 created cc: @wangsijie